### ISSUE TITLE
`npm run build` fails

### ISSUE THREAD

**Issue by Rubix982** [binodnepali/nextjs-with-material-ui-example-template#2]
Hello!

I cloned the repository and ran `yarn run build`, but it fails. I see the following logs,

```text
❯ yarn run build
yarn run v1.22.15
$ next build
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db
> Using external babel configuration
> Location: "/home/saif/Projects/Bloggr/frontend/.babelrc"
Creating an optimized production build  

Compiled successfully.


> Build error occurred
ReferenceError: test is not defined
    at Module.T/rY (/home/saif/Projects/Bloggr/frontend/.next/server/static/6uFcoKF60I8Fw8j271IE_/pages/index.spec.js:159:1)
    at __webpack_require__ (/home/saif/Projects/Bloggr/frontend/.next/server/static/6uFcoKF60I8Fw8j271IE_/pages/index.spec.js:23:31)
    at Object.4 (/home/saif/Projects/Bloggr/frontend/.next/server/static/6uFcoKF60I8Fw8j271IE_/pages/index.spec.js:99:18)
    at __webpack_require__ (/home/saif/Projects/Bloggr/frontend/.next/server/static/6uFcoKF60I8Fw8j271IE_/pages/index.spec.js:23:31)
    at /home/saif/Projects/Bloggr/frontend/.next/server/static/6uFcoKF60I8Fw8j271IE_/pages/index.spec.js:91:18
    at Object.<anonymous> (/home/saif/Projects/Bloggr/frontend/.next/server/static/6uFcoKF60I8Fw8j271IE_/pages/index.spec.js:94:10)
    at Module._compile (internal/modules/cjs/loader.js:955:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:991:10)
    at Module.load (internal/modules/cjs/loader.js:811:32)
    at Function.Module._load (internal/modules/cjs/loader.js:723:14) {
  type: 'ReferenceError'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Help is needed here. I'm unable to debug where the source of the exception arises from. The following is the content of `index.spec.js`,

```javascript
import React from "react";
import { render } from "@testing-library/react";
import Index from "./index";

test("renders deploy link", () => {
  const { getByText } = render(<Index />);
  const linkElement = getByText(
    /Instantly deploy your Next\.js site to a public URL with Vercel\./
  );
  expect(linkElement).toBeInTheDocument();
});
```

I replaced this with,

```javascript
import React from "react";
import { render } from "@testing-library/react";
import Index from "./index";

test("renders deploy link", () => {
  const { getByText } = render(<Index />);
  const linkElement = getByText(
    /Next.js Material UI Template/
  );
  expect(linkElement).toBeInTheDocument();
});
```

Because when I run the command `yarn run dev`, I saw the following page,

![image](https://user-images.githubusercontent.com/41635766/139806362-63943660-f3b0-498d-a323-c0457124a45a.png)

I am not sure why the `dev` is successful, and why the `build` is failing. It has something to do with the `tests` directory, I think.

Much thanks for the time to set this up!

**Comment by binodnepali** [comment]
Hello!
The project uses npm packager manager. I am not sure if it works with yarn. Let me know, if you have still issue when you use with npm. I will soon update the project with next.js version 11 or higher.

**Comment by Rubix982** [comment]
Hi, @binodnepali! 

Thanks for the reply!

No, for me it does not work with `npm` either. These are the commands I ran.

```bash
$ git clone https://github.com/binodnepali/nextjs-material-ui-template
$ cd nextjs-material-ui-template
$ npm install
$ npm run build
```

The build falls and throws the error,

```text
❯ npm run build

> nextjs-material-ui-template@0.1.0 build /home/saif/Projects/test/nextjs-material-ui-template
> next build

Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db
> Using external babel configuration
> Location: "/home/saif/Projects/test/nextjs-material-ui-template/.babelrc"
Creating an optimized production build  

Compiled successfully.


> Build error occurred
ReferenceError: test is not defined
    at Module.T/rY (/home/saif/Projects/test/nextjs-material-ui-template/.next/server/static/qFaVpFJI9a3DT3cvbbN1N/pages/index.spec.js:159:1)
    at __webpack_require__ (/home/saif/Projects/test/nextjs-material-ui-template/.next/server/static/qFaVpFJI9a3DT3cvbbN1N/pages/index.spec.js:23:31)
    at Object.4 (/home/saif/Projects/test/nextjs-material-ui-template/.next/server/static/qFaVpFJI9a3DT3cvbbN1N/pages/index.spec.js:99:18)
    at __webpack_require__ (/home/saif/Projects/test/nextjs-material-ui-template/.next/server/static/qFaVpFJI9a3DT3cvbbN1N/pages/index.spec.js:23:31)
    at /home/saif/Projects/test/nextjs-material-ui-template/.next/server/static/qFaVpFJI9a3DT3cvbbN1N/pages/index.spec.js:91:18
    at Object.<anonymous> (/home/saif/Projects/test/nextjs-material-ui-template/.next/server/static/qFaVpFJI9a3DT3cvbbN1N/pages/index.spec.js:94:10)
    at Module._compile (internal/modules/cjs/loader.js:955:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:991:10)
    at Module.load (internal/modules/cjs/loader.js:811:32)
    at Function.Module._load (internal/modules/cjs/loader.js:723:14) {
  type: 'ReferenceError'
}
Automatically optimizing pages .npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! nextjs-material-ui-template@0.1.0 build: `next build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the nextjs-material-ui-template@0.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/saif/.npm/_logs/2021-11-03T15_29_57_347Z-debug.log
```

My `npm` version is `6.14.15`, and my `node` version is `v12.14.1`. I switched to the version `v14.18.1` using `nvm` to no avail.

Thanks again!

BTW, some warnings generated by `npm install`,

```text
❯ npm install
npm WARN old lockfile 
npm WARN old lockfile The package-lock.json file was created with an old version of npm,
npm WARN old lockfile so supplemental metadata must be fetched from the registry.
npm WARN old lockfile 
npm WARN old lockfile This is a one-time fix-up, please be patient...
npm WARN old lockfile 

up to date, audited 1472 packages in 32s

62 packages are looking for funding
  run `npm fund` for details

62 vulnerabilities (4 low, 13 moderate, 45 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
npm notice 
npm notice New minor version of npm available! 8.0.0 -> 8.1.2
npm notice Changelog: https://github.com/npm/cli/releases/tag/v8.1.2
npm notice Run npm install -g npm@8.1.2 to update!
npm notice 
```

**Comment by binodnepali** [comment]
Hi @Rubix982 ,

I have fixed an issue. Let me know if it works for you. Run scripts using npm.
